envi/codeflow.py

```diff
@@ -261,6 +261,9 @@ def addEntryPoint(self, va, arch=envi.ARCH_DEFAULT):
             cf.addEntryPoint( 0x77c70308 )
             ... callbacks flow along ...
         '''
+        # Architecture gets to decide on actual final VA and Architecture (ARM/THUMB/etc...)
+        va, arch = self._mem.arch.archModifyFuncAddr(va, arch)
+
```

Comment:
np.  i can do this.  my initial concern is just the additional overhead of managing a dict.  arch is easy.  every opcode basically has the arch in it, every LOC_OP location as well.